server/src/main/java/com/epam/aidial/core/server/service/ResourceOperationService.java

```diff
@@ -82,4 +85,30 @@ public void moveResource(ResourceDescriptor source, ResourceDescriptor destinati
             resourceService.deleteResource(source, EtagHeader.ANY);
         }
     }
+
+    public boolean deleteResource(ResourceDescriptor resource, EtagHeader etag) {
+        String bucketName = resource.getBucketName();
```

Comment:
Make sure it takes only 4 types: conversations/prompts/files/applications

Reply:
fixed